midp/gfx.js

```diff
@@ -112,8 +112,35 @@ var currentlyFocusedTextEditor;
         return 1;
     };
 
+    var hasNewFrame = true;
+    var ctxToExecuteOnNewFrame;
+    function gotNewFrame(timestamp) {
+        if (ctxToExecuteOnNewFrame) {
+            var ctx = ctxToExecuteOnNewFrame;
+            ctxToExecuteOnNewFrame = null;
+            window.requestAnimationFrame(gotNewFrame);
+            ctx.execute();
+        } else {
+            hasNewFrame = true;
+        }
+    }
 
-    addUnimplementedNative("com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V");
+    Native["com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V"] = function() {
+        // This is a no-op: The foreground display gets drawn directly
+        // to the screen.
+        //
+        // However, we sometimes "draw" more frequently than the
+        // screen can update. By calling `requestAnimationFrame` here,
+        // we ensure that whatever we just drew will stick around for
+        // one frame
+        if (hasNewFrame) {
+            hasNewFrame = false;
+            window.requestAnimationFrame(gotNewFrame);
+        } else {
+            ctxToExecuteOnNewFrame = $.ctx;
+            $.pause(asyncImplStringAsync);
```

Comment:
Sorry for the chain of comments, but I think using execute() here is actually okay because we really do want to begin execution right away on the callback for the request animation frame and not wait until the next tick that asyncImpl would cause: Perhaps:
```javascript
var ctx = $.ctx;
$.pause(asyncImplStringAsync);
window.requestAnimationFrame(function() {
  ctx.execute();
});
```